src/backend/PhotoAPI/BusinessLogic/Services/PhotoService.cs

```diff
@@ -14,69 +14,93 @@
-
-                createdPhotos.Add(photoDTO);
-            }
-
-            return createdPhotos;
-        }
-
-        // TODO: remove this
-        private void SetPhotoValues(PhotoDocument photoDocument, string blobName)
-        {
-            photoDocument.UserId = _authService.GetCurrentUserId();
-
-            photoDocument.OriginalBlobName = blobName;
-            photoDocument.BlobName = blobName;
-            photoDocument.Blob64Name = blobName;
-            photoDocument.Blob256Name = blobName;
-        }
-
-        public Task MarkPhotosAsDeletedAsync(IEnumerable<PhotoToDeleteRestoreDTO> photosToDelete)
-        {
-            return _elasticService.MarkPhotosAsDeletedAsync(photosToDelete);
-        }
-
-    }
+	public class PhotoService : Abstract.PhotoServiceBase, IPhotoService
+	{
+		public PhotoService(
+			IMapper mapper,
+			IAuthService authService,
+			IElasticService elasticService,
+			IPhotoBlobStorage blobStorage)
+			: base(mapper, authService, elasticService, blobStorage) { }
+
+		public async Task<IEnumerable<PhotoListDTO>> GetPhotosAsync(string userId, string searchPayload)
+		{
+			IEnumerable<PhotoDocument> userPhotos = await _elasticService.GetPhotosAsync(userId, searchPayload);
+
+			return userPhotos.OrderByDescending(p => p.UploadDate).Select(_mapper.Map<PhotoListDTO>);
+		}
+
+		public async Task<PhotoViewDTO> GetPhotoOrDefaultAsync(Guid photoId)
+		{
+			PhotoDocument userPhoto = await _elasticService.GetPhotoOrDefaultAsync(photoId);
+			if (userPhoto == null) return null;
+
+			return _mapper.Map<PhotoViewDTO>(userPhoto);
+		}
+
+		public async Task<OriginalPhotoDTO> GetOriginalPhotoAsync(Guid photoId)
+		{
+			PhotoDocument userPhoto = await _elasticService.GetPhotoOrDefaultAsync(photoId);
+			if (userPhoto == null) return null;
+
+			return _mapper.Map<OriginalPhotoDTO>(userPhoto);
+		}
+
+		public async Task<IEnumerable<PhotoListDTO>> UploadPhotosAsync(IEnumerable<PhotoToUploadDTO> photosToUploadDTO)
+		{
+			List<PhotoListDTO> createdPhotos = new List<PhotoListDTO>();
+
+			foreach (PhotoToUploadDTO photo in photosToUploadDTO)
+			{
+				PhotoDocument photoDocument = _mapper.Map<PhotoDocument>(photo);
+				await SetPhotoValues(photoDocument, photo.Base64Image);
+
+				PhotoDocument createdPhoto = await _elasticService.CreateAsync(photoDocument);
+				PhotoListDTO photoDTO = _mapper.Map<PhotoListDTO>(createdPhoto);
+
+				createdPhotos.Add(photoDTO);
+			}
+
+			return createdPhotos;
+		}
+
+		public Task UpdatePhotoAsync(UpdatePhotoDTO updatePhotoDTO)
+		{
+			return _elasticService.UpdatePhotoAsync(updatePhotoDTO.Id, updatePhotoDTO);
+		}
+
+		public async Task EditPhotoAsync(EditPhotoDTO editPhotoDTO)
+		{
+
+			PhotoDocument photoDocument = await _elasticService.GetPhotoOrDefaultAsync(editPhotoDTO.Id);
+			await ClearAllBlobsExceptOriginalIfExistsAsync(new PhotoDocument[] { photoDocument });
+			await SetUpdatePhotoValues(photoDocument, editPhotoDTO.Base64Image);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add '.ConfigureAwait(false)' to this call to allow execution to continue in any thread.](https://app.codacy.com/manual/iamprovidence/Lama/pullRequest?prid=5141404)